AppInspector.RulesEngine/RulesVerifier.cs

```diff
@@ -148,12 +148,27 @@ public RuleStatus CheckIntegrity(ConvertedOatRule convertedOatRule)
                     {
                         throw new ArgumentException();
                     }
-
+                    
                     RegexOptions regexOpts =
-                        Utils.RegexModifierToRegexOptions(searchPattern.Modifiers ?? Array.Empty<string>());
+                        Utils.RegexModifierToRegexOptions(searchPattern.Modifiers);
+                    
+#if NET7_0_OR_GREATER
+                    if (_options.EnableNonBacktrackingRegex)
+                    {
+                        regexOpts |= RegexOptions.NonBacktracking;
+                    }
+#endif
 
                     _ = new Regex(searchPattern.Pattern, regexOpts);
                 }
+                catch (NotSupportedException e)
+                {
+                    // Regex requires backtracking, try without non-backtracking
+                    _logger?.LogWarning(MsgHelp.GetString(MsgHelp.ID.VERIFY_RULES_REGEX_FAIL), rule.Id ?? "", searchPattern.Pattern ?? "", e.Message);
+                    errors.Add(MsgHelp.FormatString(MsgHelp.ID.VERIFY_RULES_REGEX_FAIL, rule.Id ?? "", searchPattern.Pattern ?? "", e.Message));
+
+                    _ = new Regex(searchPattern.Pattern, Utils.RegexModifierToRegexOptions(searchPattern.Modifiers));      
```

Comment:
Should use regexOpts from before with

`regexOpts &= ~RegexOptions.NonBackTracking` right?

Reply:
regexOpts won't be available here due to the declaration scope. We will need to move it above `try` block, but we might have some exceptions from RegexModifierToRegexOptions method. So we will need to put it to another try.
Not sure if we are catching exceptions "higher".

Reply:
I don't think `RegexModifierToRegexOptions` has any cases where it will throw, so it should be safe to move out of the try.

Reply:
Haha, you are right. For some reason I read the code below the actual method in the Utils.cs file. Will change!